src/mysql.c

```diff
@@ -1014,6 +1091,8 @@ static int mysql_read (user_data_t *ud)
 
 		gauge_submit ("cache_size", "qcache",
 				qcache_queries_in_cache, db);
+ 		gauge_submit ("cache_queries", "qcache-free-memory",
```

Comment:
The *memory* type is for bytes held in memory, which is *exactly* what this metric is afaIk. If you're afraid that this might clash with memory metrics that are not related to the query cache, let's use the *type instance* "qcache-free".

Adding a new type "cache_memory" is not justified.